internal/rbd/encryption.go

```diff
@@ -135,8 +135,11 @@ func (ri *rbdImage) copyEncryptionConfig(cp *rbdImage) error {
 		return fmt.Errorf("failed to fetch passphrase for %q: %w",
 			ri, err)
 	}
-
-	cp.encryption, err = util.NewVolumeEncryption(ri.encryption.GetID(), ri.encryption.KMS)
+	if copyOnlyPassphrase {
+		cp.encryption, err = util.NewVolumeEncryption(cp.encryption.GetID(), cp.encryption.KMS)
+	} else {
+		cp.encryption, err = util.NewVolumeEncryption(ri.encryption.GetID(), ri.encryption.KMS)
+	}
```

Comment:
instead of this new variable, the same can be done if we compare KMS/ID?

Reply:
Actually cp.encrypyion is `nil` at places, so we cannot do that.
(Since it is reused for snapshotting and cloning (+intermediate steps))